sdcm/cluster_docker.py

```diff
@@ -52,12 +52,18 @@ def node_container_run_args(self, seed_ip):
         volumes = {
             '/var/run/docker.sock': {"bind": '/var/run/docker.sock', "mode": "rw"},
         }
+
+        # TODO: add parsing of cpuset option from append_scylla_args parameter
+        scylla_args = self.parent_cluster.params.get('append_scylla_args')
+        smp_match = re.search(r'--smp\s(\d+)', scylla_args)
```

Comment:
This solution looks not optimal.

Maybe do following:
- Read number of cores (CORE_NUM) on the node where docker runs
- Read "n_db_nodes" value (i.e. '3') add "1" (CONT_NUM) to allow to run "extend cluster" kind of nemesis.
- Decide what is the CPU usage per container by diving CORE_NUM by CONT_NUM
  * if it is less than 2, apply 1.
  * if it is not round like "2.0" then use integer part.

Reply:
My proposal is based exactly on the idea of all-in one solution.

Reply:
I don't think we should do calculation for this case, it would be really problematic, and we won't know the exact setup that was used for a passing test.

there's not real need for it to be "production" like, just that we'll have reference of cases that works for our needs
and that we can configure what needed

Reply:
I think we are o.k. with limiting this solution to db nodes only for now.

We need to adapt cases to docker backend anyhow regardless, cause of multiple reasons.

If we'll found we need a way to define number of CPU for loader, we should add it to SCT configuration and align it

Reply:
Changed the current logic to set smp value from scylla_args for db-nodes only. Other nodes get smp=1. 

This is an example of running this locally:
```
❯ grep smp configurations/nemesis/additional_configs/docker_backend_local.yaml
append_scylla_args: '--smp 4 --memory 2G'
❯ docker ps -q | xargs -I {} docker inspect --format='{{.Name}}: {{.HostConfig.NanoCpus}}' {} | awk '{print $1 ": " $2 / 1000000000 " CPUs"}'

/cool_allen:: 0 CPUs
/agraf:: 0 CPUs
/agrafrender:: 0 CPUs
/aprom:: 0 CPUs
/promtail:: 0 CPUs
/loki:: 0 CPUs
/aalert:: 0 CPUs
/longevity-1gb-1h-nemesis-dmitriy-loader-node-949e8078-0:: 1 CPUs
/longevity-1gb-1h-nemesis-dmitriy-db-node-949e8078-2:: 4 CPUs
/longevity-1gb-1h-nemesis-dmitriy-db-node-949e8078-1:: 4 CPUs
/longevity-1gb-1h-nemesis-dmitriy-db-node-949e8078-0:: 4 CPUs
```

As an observation, even with the rate limited stress commands for docker backend CPU utilization by db nodes is higher than 1. So the change can be helpful if we want to avoid unintentional throttling of db nodes performance on docker backend.
```
docker stats --no-stream --format "table {{.Name}}\t{{.CPUPerc}}\t{{.MemUsage}}\t{{.MemPerc}}"
NAME                                                      CPU %     MEM USAGE / LIMIT   MEM %
wizardly_montalcini                                       78.31%    119.8MiB / 31GiB    0.38%
friendly_taussig                                          44.65%    587.8MiB / 31GiB    1.85%
modest_varahamihira                                       43.13%    564.6MiB / 31GiB    1.78%
agraf                                                     0.05%     229.2MiB / 31GiB    0.72%
agrafrender                                               0.00%     107.1MiB / 31GiB    0.34%
aprom                                                     0.00%     198.1MiB / 31GiB    0.62%
promtail                                                  0.24%     78.18MiB / 31GiB    0.25%
loki                                                      0.23%     97.41MiB / 31GiB    0.31%
aalert                                                    0.48%     39.58MiB / 31GiB    0.12%
longevity-1gb-1h-nemesis-dmitriy-loader-node-949e8078-0   0.72%     264.3MiB / 31GiB    0.83%
longevity-1gb-1h-nemesis-dmitriy-db-node-949e8078-2       142.32%   1.864GiB / 31GiB    6.01%
longevity-1gb-1h-nemesis-dmitriy-db-node-949e8078-1       167.39%   1.933GiB / 31GiB    6.24%
longevity-1gb-1h-nemesis-dmitriy-db-node-949e8078-0       266.26%   2.059GiB / 31GiB    6.64%
```